homeassistant/components/logbook/__init__.py

```diff
@@ -209,12 +217,62 @@ def _async_describe_event(
     platform.async_describe_events(hass, _async_describe_event)
 
 
+def _async_determine_event_types(
+    hass: HomeAssistant, entity_ids: list[str] | None, device_ids: list[str] | None
+) -> tuple[str, ...]:
+    """Reduce the event types based on the entity ids and device ids."""
+    external_events: dict[
+        str, tuple[str, Callable[[LazyEventPartialState], dict[str, Any]]]
+    ] = hass.data.get(DOMAIN, {})
+    if not entity_ids and not device_ids:
+        return (*ALL_EVENT_TYPES_EXCEPT_STATE_CHANGED, *external_events)
+    config_entry_ids: set[str] = set()
+    intrested_event_types: set[str] = set()
+
+    if entity_ids:
+        #
+        # Home Assistant doesn't allow firing events from
+        # entities (at least in code review)
+        # so we have a limited list to check
```

Comment:
This is not allowed. Events are tied to devices.

```suggestion
        # entities so we have a limited list to check
```